### ISSUE TITLE
Portal Blocks fail to import feeds from sites using https

### ISSUE THREAD

**Issue by mystralkk** [Geeklog-Core/geeklog#732]
When portal blocks try to import feed from sites using https, they can fail and error message left in error.log reads:

> xx.xx.xx.xx - HTTP Fetch Failed  Unable to connect to tls://example.net:443. Error: stream_socket_client(): unable to connect to tls://example.net:443 (Unknown error)
> stream_socket_client(): Failed to enable crypto
> stream_socket_client(): SSL operation failed with code 1. OpenSSL Error messages:
> error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed

I fixed this issue by:

1. Downloading certificate from https://curl.haxx.se/ca/cacert.pem and uploaded the file to $_CONF['path_data'].
2. Changing the first part of FeedParserFactory::_getFeed 
from:
>        $req = new HTTP_Request2(
>             $url,
>             HTTP_Request2::METHOD_GET,
>             array('follow_redirects' => true)
>         );
> 

to:
>         global $_CONF;
> 
>         $options = array(
>             'follow_redirects' => true
>         );
> 
>         if (stripos($url, 'https:') === 0) {
>             $options['ssl_verify_peer'] = true;
>             $options['ssl_cafile'] = $_CONF['path_data'] . 'cacert.pem';
>         }

Is this the right way?  If so, I will commit this change to the repository.

**Comment by eSilverStrike** [comment]
If HTTP_Request2 uses curl (it says it is optional http://pear.php.net/package/HTTP_Request2) then it can be set via the php.ini file:
```
Set curl.cainfo=”c:\my php path\cacert.pem”
```
This fixes curl not being able to access https. See:
http://nl3.php.net/manual/en/function.curl-setopt.php#110457
http://curl.haxx.se/docs/caextract.html

Since it is optional I assume it may do it a different way.  If we do require CURL for HTTPS then we should add a check for it on the Hosting Environment Check page similar to the OpenSLL Library check.

Maybe we can trap for this import feeds error some how and include a message which points to the Geeklog docs with the fix?

**Comment by eSilverStrike** [comment]
Is this still an issue? I was trying to reproduce the error by creating a portal block and pointing it to https://www.geeklog.net/backend/geeklog.rss and was not able to create the error on my development server which has OpenSSL enabled but no cert.perm.

On my live server I do have  the openssl.capath set in php.ini set to:
openssl.capath=C:\...\cert.pem

I could add a test in the Geeklog Environment Check to retrieve 
$info = ini_get('openssl.cafile');
and if empty add a message that it should be set and where to get the file.

Thoughts?

**Comment by mystralkk** [comment]
If you or somebody else write about this issue and solution in the Wiki, we don't have to add an extra test to the Geeklog Environment Check.

**Comment by eSilverStrike** [comment]
It looks the php ini settings where not added until php 5.3.7 (at least for curl not sure about openssl which might be php 5.4). 

So the php.ini fix will not work for those using for example php 5.3.3 which is the oldest version Geeklog supports.

**Comment by eSilverStrike** [comment]
Full update would be
```
    global $_CONF;

    $options = array(
        'follow_redirects' => true
    );

    if (stripos($url, 'https:') === 0) {
        $options['ssl_verify_peer'] = true;
        $options['ssl_cafile'] = $_CONF['path_data'] . 'cacert.pem';
    }

   $req = new HTTP_Request2(
        $url,
        HTTP_Request2::METHOD_GET,
        $options);
```

**Comment by mystralkk** [comment]
Fixed with change set bbe6321
